cmd/clusterctl/pkg/client/init.go

```diff
@@ -68,40 +71,53 @@ func (c *clusterctlClient) Init(options InitOptions) ([]Components, bool, error)
 		watchingNamespace: options.WatchingNamespace,
 	}
 
+	log.Info("Fetching providers")
+
 	if options.CoreProvider != "" {
 		if err := c.addToInstaller(addOptions, clusterctlv1.CoreProviderType, options.CoreProvider); err != nil {
-			return nil, false, err
+			return nil, err
 		}
 	}
 
 	if err := c.addToInstaller(addOptions, clusterctlv1.BootstrapProviderType, options.BootstrapProviders...); err != nil {
-		return nil, false, err
+		return nil, err
 	}
 
 	if err := c.addToInstaller(addOptions, clusterctlv1.ControlPlaneProviderType, options.ControlPlaneProviders...); err != nil {
-		return nil, false, err
+		return nil, err
 	}
 
 	if err := c.addToInstaller(addOptions, clusterctlv1.InfrastructureProviderType, options.InfrastructureProviders...); err != nil {
-		return nil, false, err
+		return nil, err
 	}
 
 	// Before installing the providers, ensure the cert-manager WebHook is in place.
 	if err := cluster.CertManger().EnsureWebHook(); err != nil {
-		return nil, false, err
+		return nil, err
 	}
 
 	components, err := installer.Install()
 	if err != nil {
-		return nil, false, err
+		return nil, err
+	}
+
+	// IF this is the firstRun, then log the usage instructions.
+	if firstRun && options.LogUsageInstructions {
+		log.Info("")
+		log.Info("Your cluster API management cluster has been initialized successfully!")
+		log.Info("")
+		log.Info("You can now create your first workload cluster by running the following:")
+		log.Info("")
+		log.Info("  clusterctl config cluster [name] --kubernetes-version [version] | kubectl apply -f -")
```

Comment:
This message seems out of place for the library, since it specifically calls out the binary name and arguments here.

Reply:
The former solution for handling this message was even worse,  with a boolean flag in output that was accepted during the review with the commitment to change this as soon as possible.

But I agree with your concern and this is why those logs are generated only if the caller explicitly requires it, and by default, they are not generated.